lib/src/file.dart

```diff
@@ -49,15 +49,28 @@ class SourceFile {
   /// previous result.
   int _cachedLine;
 
-  /// Creates a new source file from [text].
+  /// This constructor is deprecated.
   ///
-  /// [url] may be either a [String], a [Uri], or `null`.
+  /// Use [new SourceFile.fromString] instead.
+  @Deprecated("Will be removed in 2.0.0")
```

Comment:
The documentation of `@Deprecated` is pretty clear that the string is only meant to include a description of when the feature expires. That's a kind of strange choice, but given that I think it's fine to leave more detailed description to the documentation.